posydon/unit_tests/utils/test_gridutils.py

```diff
@@ -0,0 +1,760 @@
+                   MESA_data.dtype.names) is None
+
+    def test_fix_He_core(self, MESA_data):
+        # missing argument
+        with raises(TypeError, match="missing 1 required positional"+\
+                    " argument: 'history'"):
+            totest.fix_He_core()
+        # history is None
+        assert totest.fix_He_core(None) is None
+        # history is ndarray without the required columns
+        assert totest.np.array_equal(MESA_data, totest.fix_He_core(MESA_data))
+        # history is ndarray with the required columns and corrects them
+        test_history = totest.np.array([(1, 2, 3, 4), (1, 2, 4, 3),\
+                                        (2, 1, 3, 4), (2, 1, 4, 3)],\
+                                       dtype=[('he_core_mass', '<f8'),\
+                                              ('co_core_mass', '<f8'),\
+                                              ('he_core_radius', '<f8'),\
+                                              ('co_core_radius', '<f8')])
+        fixed_history = totest.np.array([(2, 2, 4, 4), (2, 2, 4, 3),\
+                                         (2, 1, 4, 4), (2, 1, 4, 3)],\
+                                        dtype=[('he_core_mass', '<f8'),\
+                                               ('co_core_mass', '<f8'),\
+                                               ('he_core_radius', '<f8'),\
+                                               ('co_core_radius', '<f8')])
+        assert totest.np.array_equal(fixed_history,\
+               totest.fix_He_core(test_history))
+
+    def test_add_field(self, MESA_data):
+        # missing argument
+        with raises(TypeError, match="missing 2 required positional"+\
+                    " arguments: 'a' and 'descr'"):
+            totest.add_field()
+        # add to empty ndarray
+        with raises(ValueError, match="'a' must be a structured numpy array"):
+            totest.add_field(totest.np.array([]), [('new', '<f8')])
+        # add to test data
+        extended_ndarray = totest.add_field(MESA_data, [('new', '<f8')])
+        assert MESA_data.dtype.descr+[('new', '<f8')] ==\
+               extended_ndarray.dtype.descr
+        assert totest.np.array_equal(MESA_data,\
+               extended_ndarray[[k for k in MESA_data.dtype.names]])
+
+    def test_get_cell_edges(self):
+        # missing argument
+        with raises(TypeError, match="missing 2 required positional"+\
+                    " arguments: 'grid_x' and 'grid_y'"):
+            totest.get_cell_edges()
+        # example for x = [0.1, 0.3, 0.5, 0.7, 0.9] and y = [0.1, 0.3, 0.5]
+        assert totest.np.allclose(\
+               totest.get_cell_edges(totest.np.linspace(0.1, 0.9, 5),\
+                                     totest.np.linspace(0.1, 0.5, 3)),\
+               totest.np.meshgrid(totest.np.linspace(0.0, 1.0, 6),\
+                                  totest.np.linspace(0.0, 0.6, 4)))
+
+    def test_find_nearest(self):
```

Comment:
This would be a good improvement for the tested code, so why don't you put the comment on the code itself (in case this needs to be addressed in this PR) or an issue up for it (in case to be outside the scope of this PR)?